eldritch/ancient_ones.py

```diff
@@ -0,0 +1,169 @@
+import abc
+
+from eldritch import events, places, characters, monsters, values
+
+
+class AncientOne(metaclass=abc.ABCMeta):
+  # pylint: disable=unused-argument
+  def __init__(self):
+    self.doom = 0
+    self.combat_difficulty = 0
+    self.attributes = []
+
+  def get_interrupts(self, owner, state):
+    return []
+
+  def get_modifier(self, thing, attribute):
+    return 0
+
+  def get_override(self, other, attribute):
+    return None
+
+  def awaken(self, state):
+    pass
+
+  @abc.abstractmethod
+  def attack(self, state):
+    raise NotImplementedError
+
+
+class DummyAncient(AncientOne):
+  def attack(self, state):
+    pass
+
+
+class SquidFace(AncientOne):
+  def __init__(self):
+    super().__init__()
+    self.name = "SquidFace"
+    self.max_doom = 13
+    self.combat_difficulty = -6
+
+  def get_modifier(self, thing, attribute):
+    if isinstance(thing, monsters.Cultist):
+      return {"horror": -2, "horror_damage": 2}.get(attribute, 0)
+    if isinstance(thing, characters.Character):
+      return {"maximum_sanity": -1, "maximum_stamina": -1}
+    return 0
+
+  def attack(self, state):
+    self.doom = min(self.doom + 1, self.max_doom)
+    # TODO: Each character lowers max sanity or stamina
+
+
+class YellowKing(AncientOne):
+  def __init__(self):
+    super().__init__()
+    self.name = "The Yellow King"
+    self.attributes = ["physical resistance"]
+    self.combat_difficulty = None
+    self.luck_modifier = 1
+    self.max_doom = 13
+
+  def awaken(self, state):
+    self.combat_difficulty = state.terror
+
+  def get_modifier(self, thing, attribute):
+    if isinstance(thing, monsters.Cultist):
+      return {"combat_difficulty": -2, "movement": "flying"}.get(attribute, 0)
+    if isinstance(thing, events.GateCloseAttempt):
+      return 3
+    return 0
+
+  def attack(self, state):
+    checks = []
+    for char in state.characters:
+      check = events.Check(char, "luck", self.luck_modifier)
+      checks.append(
+          events.PassFail(char, check, events.Nothing(), events.Loss(char, {"sanity": 2}))
+      )
+    state.event_stack.append(events.Sequence(checks))
+    self.luck_modifier -= 1
+
+
+class ChaosGod(AncientOne):
+  def __init__(self):
+    super().__init__()
+    self.name = "God of Chaos"
+    self.max_doom = 14
+
+  def awaken(self, state):
+    state.game_stage = "defeat"
+
+  def attack(self, state):
+    pass
+
+  def get_modifier(self, thing, attribute):
+    if isinstance(thing, monsters.Maniac):
+      return {"toughness": 1}.get(attribute, 0)
+    return 0
+
+
+class Wendigo(AncientOne):
+  def __init__(self):
+    super().__init__()
+    self.name = "Wendigo"
+    self.max_doom = 11
+    self.combat_difficulty = -3
+    self.fight_modifier = 1
+
+  def get_interrupts(self, owner, state):
+    if (
+        isinstance(owner, events.CityMovement)
+        and owner.is_resolved()
+        and isinstance(owner.character.place, places.Street)
```

Comment:
This should also have a test.

Reply:
To be more specific, probably a test that inherits from EventTest, has a character in the street, and then advance to the mythos phase and then the next upkeep. Validate that the character loses a stamina.
Note that this ancient one's ability applies to the mythos phase, not the movement phase. So you'll want to add a test to make sure a character that is in a location during the movement phase but gets kicked out during the encounter phase still loses one stamina.
Lastly, you'll want to have a negative test that validates that a character in a location does not lose one stamina.

Reply:
Done.